cmd/src/campaigns_common.go

```diff
@@ -290,6 +286,34 @@ func campaignsExecute(ctx context.Context, out *output.Output, svc *campaigns.Se
 	return id, url, nil
 }
 
+// campaignsParseSpec parses and validates the given campaign spec. If the spec
+// has validation errors, the errors are output in a human readable form and an
+// exitCodeError is returned.
+func campaignsParseSpec(out *output.Output, svc *campaigns.Service, input io.ReadCloser) (*campaigns.CampaignSpec, string, error) {
+	spec, raw, err := svc.ParseCampaignSpec(input)
+	if err != nil {
+		if merr, ok := err.(*multierror.Error); ok {
+			block := out.Block(output.Line("\u274c", output.StyleWarning, "Campaign spec failed validation."))
+			defer block.Close()
+
+			for i, err := range merr.Errors {
+				block.Writef("%d. %s", i+1, err)
+			}
+
+			return nil, "", &exitCodeError{
+				error:    nil,
+				exitCode: 2,
+			}
+		} else {
+			// This shouldn't happen; let's just punt and let the normal
+			// rendering occur.
+			return nil, "", err
+		}
+	}
+
+	return spec, raw, nil
+}
```

Comment:
Just wondering if we should incorporate this logic into `ParseCampaignSpec`, then we wouldn't need this function that gets passed in a service, and it can't be forgotten. Are there use-cases where we explicitly would not want to have this error prettify logic?

Reply:
It's basically a layering thing: I don't want `internal/campaigns` (in this case, the `Service` methods) to have any specific knowledge of `internal/output`, so this code lives here to basically handle the rendering of any errors. This function is used by each of the campaign commands that have to parse a spec file, though, so it's still shared.